launcher/mainwindow_moc.cpp

```diff
@@ -130,21 +130,42 @@ MainWindow::MainWindow(QWidget * parent)
 
 void MainWindow::detectPreferredLanguage()
 {
-	auto preferredLanguages = QLocale::system().uiLanguages();
+	// Skip autodetection if language is already set to a valid VCMI language
+	std::string currentLanguage = settings["general"]["language"].String();
+
+	if (!currentLanguage.empty())
+	{
+		for (const auto &vcmiLang : Languages::getLanguageList())
+		{
+			if (vcmiLang.identifier == currentLanguage && vcmiLang.selectable)
+			{
+				logGlobal->info("Language '%s' is already valid, skipping autodetection", currentLanguage);
+				return;
+			}
+		}
+	}
+
+	// Proceed with autodetection
+	auto appLanguages = QLocale().uiLanguages();
+	auto sysLanguages = QLocale::system().uiLanguages();
```

Comment:
Will this really result in different lists?

https://doc.qt.io/qt-6/qlocale.html#QLocale

> If no default locale was set using [setDefault](https://doc.qt.io/qt-6/qlocale.html#setDefault)(), this locale will be the same as the one returned by [system](https://doc.qt.io/qt-6/qlocale.html#system)().